http/client.go

```diff
@@ -18,23 +20,49 @@ import (
 
 var ErrInvalidMethod = errors.New("webhook only supports HTTP methods PUT or POST")
 
-type ClientConfiguration struct {
-	UserAgent string
-}
-
-var DefaultClientConfiguration = ClientConfiguration{
-	UserAgent: "Grafana",
+type clientConfiguration struct {
+	userAgent string
+	dialer    net.Dialer // We use Dialer here instead of DialContext as our mqtt client doesn't support DialContext.
 }
 
 type Client struct {
-	log   logging.Logger
-	agent string
+	log logging.Logger
+	cfg clientConfiguration
 }
 
-func NewClient(log logging.Logger, cfg ClientConfiguration) *Client {
+func NewClient(log logging.Logger, opts ...ClientOption) *Client {
+	cfg := clientConfiguration{
+		userAgent: "Grafana",
+		dialer: net.Dialer{
+			Timeout: 30 * time.Second,
+		},
+	}
+	for _, opt := range opts {
+		if opt != nil {
+			opt(&cfg)
+		}
+	}
 	return &Client{
-		log:   log,
-		agent: cfg.UserAgent,
+		log: log,
+		cfg: cfg,
+	}
+}
+
+func (ns *Client) Dialer() net.Dialer {
+	return ns.cfg.dialer
+}
```

Comment:
Is there a plan to use this method in other repos? I couldn't find any calls here.

Reply:
It's used in the followup PR. Accidentally left it in this one.